src/main/java/com/mongodb/kafka/connect/sink/processor/field/projection/BlockListProjector.java

```diff
@@ -40,14 +40,7 @@ protected void projectDocument(final BsonDocument doc) {
   }
 
   private void doProjection(final String field, final BsonDocument doc) {
-    if (!field.contains(FieldProjector.SUB_FIELD_DOT_SEPARATOR)) {
-      if (field.equals(FieldProjector.SINGLE_WILDCARD)
-          || field.equals(FieldProjector.DOUBLE_WILDCARD)) {
-        handleWildcard(field, "", doc);
-        return;
-      }
-
-      doc.remove(field);
+    if (processNonNestedMatches(field, doc)) {
```

Comment:
[nit] I usually find that having multiple return statements increases cognitive complexity. It seems that processNonNestedMatches returns false only when !field.contains(FieldProjector.SUB_FIELD_DOT_SEPARATOR) is false. If we move this check before entering the method, we might slightly simplify the code by avoiding return statements inside the method.

if (!field.contains(FieldProjector.SUB_FIELD_DOT_SEPARATOR)) {
        processNonNestedMatches(field, doc);
    }

Reply:
Done